src/acl/FilledChecklist.cc

```diff
@@ -233,8 +239,13 @@ ACLFilledChecklist::ACLFilledChecklist(const acl_access *A, HttpRequest *http_re
         if (request->clientConnectionManager.valid())
             conn(request->clientConnectionManager.get());
     }
+}
 
+void
+ACLFilledChecklist::setIdent(const char *ident)
+{
 #if USE_IDENT
+    assert(!rfc931[0]);
```

Comment:
This assert does not look right. It is ensuring that the setIdent() can only overwrite previous values, and never sets the first one.

Reply:
@yadij, please take a second look. I believe the code (intensionally) does the opposite of what you say it does -- it checks that setIdent() is not called for an already "configured" checklist. In other words, we do not want to support multiple/repeated setIdent() calls. This check does not catch all violations, but it is cheap and shows the intent well, so I would keep it.

Reply:
Sorry, my bad wording. The "rfc931[0]" part of the condition is a nullptr de-reference unless the char* has already been allocated/assigned previously.

Reply:
> The "rfc931[0]" part of the condition is a nullptr de-reference

Please take a third look: FilledChecklist:rfc931 is not a pointer.

```
```